cogs/cryochamber.py

```diff
@@ -0,0 +1,108 @@
+"""Commands for preserving Discord messages in another channel."""
+import re
+
+from discord import Embed, Colour
+from discord.ext.commands import (Cog, CommandError, MessageConverter,
+                                  TextChannelConverter, group, check)
+
+
+class Cryochamber(Cog):
+    def __init__(self, _):
+        self.ongoing_preservations = set()
+
+    async def can_manage_messages(ctx):
+        get_channel = TextChannelConverter().convert
+        arg = ctx.message.content
+        exp = r'(?P<source_specifier>.*?) in (?P<destination_channel_specifier>.*?)'
+        command_match = re.fullmatch(exp, arg)
+        dest_channel = await get_channel(ctx,
+                                         command_match.group('destination_channel_specifier'))
+        if not dest_channel.permissions_for(ctx.author).manage_messages:
+            return False
+        else:
+            return True
+
+    """Main preserve function. Things in [] are optional arguments, stuff in () are variables.
```

Comment:
In order to make this the `?help` text in the bot, it needs to be set as the function docstring by putting it immediately under the function declaration. Discord.py is smart enough to include this as help, and the first line will become the "short form" help shown in the main command list.